sagemaker-python-sdk/tensorflow_script_mode_using_shell_commands/tensorflow_script_mode_using_shell_commands.ipynb

```diff
@@ -0,0 +1,349 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import Image\n",
+    "Image(filename='training/resting-dragon.jpg') "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the training script\n",
+    "\n",
+    "Let's start by downloading the [deep_dream](https://github.com/keras-team/keras/blob/2.2.4/examples/deep_dream.py) example script from Keras repository. This script takes and image an uses deep dream algorithm to generate\n",
+    "transformations of that image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget https://raw.githubusercontent.com/keras-team/keras/2.2.4/examples/deep_dream.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The script **deep_dream.py** takes two positional arguments:\n",
+    "- `base_image_path`: Path to the image to transform.\n",
+    "- `result_prefix`: Prefix of all generated images.\n",
+    "\n",
+    "### Creating the launcher script\n",
+    "\n",
+    "We need to create a launcher script that installs Keras, sets `base_image_path` \n",
+    "and `result_prefix`, and invokes **deep_dream.py**:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile launcher.sh \n",
+    "\n",
+    "pip install keras==2.2.4\n",
+    "\n",
+    "BASE_IMAGE_PATH=\"${SM_CHANNEL_TRAINING}/resting-dragon.jpg\"\n",
+    "RESULT_PREFIX=\"${SM_MODEL_DIR}/dream\"\n",
+    "\n",
+    "python deep_dream.py ${BASE_IMAGE_PATH} ${RESULT_PREFIX}\n",
+    "\n",
+    "echo \"Generated image $(ls ${SM_MODEL_DIR})\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**SM_CHANNEL_TRAINING** and **SM_MODEL** are environment variables created by the SageMaker TensorFlow\n",
+    "Container in the beginning of training. Let's take a more detailed look at then: \n",
+    "\n",
+    "- **SM_MODEL_DIR**: the directory inside the container where the training model data must be saved inside the container, i.e. /opt/ml/model.\n",
+    "- **SM_TRAINING_CHANNEL**: the directory containing data in the 'training' channel. \n",
+    "\n",
+    "For more information about training environment variables, please visit [SageMaker Containers](https://github.com/aws/sagemaker-containers#list-of-provided-environment-variables-by-sagemaker-containers)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test locally using SageMaker Python SDK TensorFlow Estimator\n",
+    "You can use the SageMaker Python SDK TensorFlow estimator to easily train locally and in SageMaker.\n",
+    "Let's start setting **launcher.sh** as the entry-point and **deep_dream.py** as a dependency:"
```

Comment:
Let's set **launcher.sh** as the ....

Reply:
I think it's helpful to have a reference here to the documentation of the estimator specifically the part taking out entry_point and dependencies